File: js/src/tests/test_base.js
// Copyright (c) 2015-2017, Exa Analytics Development Team
// Distributed under the terms of the Apache License 2.0
/**
 * Generic Widget Tests
 */
"use strict";
var _ = require("underscore");
//var ipy = require("@jupyter-widgets/base");    // >=7.x
var ipy = require("jupyter-js-widgets");    // <=6.x
var base = require("../base.js");


var DOMWidgetTestModel = ipy.DOMWidgetModel.extend({
    defaults: _.extend({}, ipy.DOMWidgetModel.prototype.defaults, base.defaults, {
        _model_name: "DOMWidgetTestModel",
        _view_name: "DOMWidgetTestView",
        value: "Hello World"
    })
});


var DOMWidgetTestView = ipy.DOMWidgetView.extend({
    render: function() {
        this.value_changed();
        this.model.on("change:value", this.value_changed, this);
    },

    value_changed: function() {
        console.log(this);
        console.log(this.el);
        this.el.textContent = this.model.get("value");
    }
});


/**
 * Test non DOM widgets
 */
var WidgetTestModel = ipy.WidgetModel.extend({
    defaults: _.extend({}, ipy.WidgetModel.prototype.defaults, base.defaults, {
        _model_name: "WidgetTestModel",
        _view_name: "WidgetTestView"
    })
});


/**
 * Test non DOM Widgets
 */
var WidgetTestView = ipy.WidgetView.extend({
    render: function() {
        this.log_value();
        this.send_value();
    },

    log_value: function() {
        var value = this.model.get("value");
        console.log(value);
    },

    send_value: function() {
        var value = this.model.get("value");
        this.send(value);
    }
});


module.exports = {
    WidgetTestModel: WidgetTestModel,
    WidgetTestView: WidgetTestView,
    DOMWidgetTestModel: DOMWidgetTestModel,
    DOMWidgetTestView: DOMWidgetTestView
};

File: exa/widgets/tests/test_base.py
# -*- coding: utf-8 -*-
# Copyright (c) 2015-2017, Exa Analytics Development Team
# Distributed under the terms of the Apache License 2.0
"""
Test Base Widgets
#############################################
"""
#import os, platform
#from traitlets import Unicode
#from unittest import TestCase
#from ipywidgets import DOMWidget, Widget
#from subprocess import check_output
#from exa.widgets import jsver, jsmod
#
#
#nbname = "test_base.ipynb"
#cwd = os.path.dirname(os.path.abspath(__file__))
#nbpath = os.path.join(cwd, nbname)
#tmppath = os.path.join(cwd, "test_base.ipynb.output")
#htmlpath = os.path.join(cwd, nbname.replace(".ipynb", ".html"))
#prckws = {'shell': True} if platform.system().lower() == "windows" else {}
#
#
#class WidgetTest(Widget):
#    """
#    """
#    _view_module = Unicode(jsmod).tag(sync=True)
#    _view_module_version = Unicode(jsver).tag(sync=True)
#    _view_name = Unicode("WidgetTestView").tag(sync=True)
#    _model_module = Unicode(jsmod).tag(sync=True)
#    _model_module_version = Unicode(jsver).tag(sync=True)
#    _model_name = Unicode("WidgetTestModel").tag(sync=True)
#    value = Unicode("WidgetTest value").tag(sync=True)
#    responded = False
#
#    def set_value_response(self, *args, **kwargs):
#        """
#        """
#        WidgetTest.responded = True
#
#    def __init__(self, *args, **kwargs):
#        super(WidgetTest, self).__init__(*args, **kwargs)
#        self.on_msg(self.set_value_response)
#
#
#class DOMWidgetTest(DOMWidget):
#    """
#    """
#    _view_module = Unicode(jsmod).tag(sync=True)
#    _view_module_version = Unicode(jsver).tag(sync=True)
#    _view_name = Unicode("DOMWidgetTestView").tag(sync=True)
#    _model_module = Unicode(jsmod).tag(sync=True)
#    _model_module_version = Unicode(jsver).tag(sync=True)
#    _model_name = Unicode("DOMWidgetTestModel").tag(sync=True)
#    value = Unicode("DOMWidgetTest value").tag(sync=True)
#
#
#class TestWidgets(TestCase):
#    """
#    """
#    pass
##    def setUp(self):
##        """
##        """
##        ret = check_output(["jupyter", "nbconvert", "--exec",
##                            "--ExecutePreprocessor.kernel_name=python", nbpath],
##                           cwd=cwd, **prckws)
##        self.expected = ["False", "True", "DOMWidgetTest value", "test", "test"]
##        self.expected_nodom = self.expected[:]
##        self.expected_nodom[1] = "False"
##
##    def test_value_check(self):
##        """
##        """
##        try:
##            with open(tmppath) as f:
##                result = [line.strip() for line in f.readlines()]
##            check = result == self.expected
##            if check == False:
##                check = result == self.expected_nodom
##            self.assertTrue(check)
##        except Exception as e:
##            self.fail(msg=str(e))
##
##    def tearDown(self):
##        """
##        """
##        try:
##            os.remove(tmppath)
##            os.remove(htmlpath)
##        except Exception as e:
##            self.fail(msg=str(e))
